File: VB/PersonalOrganizer/ContactContextDataModel/DesignTime/DesignTimeContactContextUnitOfWork.vb
Imports System
Imports System.Linq
Imports System.Data
Imports System.Data.Entity
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports Common.Utils
Imports Common.DataModel
Imports Common.DataModel.EntityFramework
Imports PersonalOrganizer.Model
Namespace ContactContextDataModel
    Public Class ContactContextDesignTimeUnitOfWork
        Inherits DesignTimeUnitOfWork
        Implements IContactContextUnitOfWork
        Private Shared _contactsRepository As DesignTimeContactRepository = New DesignTimeContactRepository()
        Public Sub New()
        End Sub
        Private Function HasChanges() As Boolean Implements IContactContextUnitOfWork.HasChanges
            Return False
        End Function
        Private ReadOnly Property Contacts As IContactRepository Implements IContactContextUnitOfWork.Contacts
            Get
                Return _contactsRepository
            End Get
        End Property
        Public Sub Detach(ByVal entity As Object)
        End Sub
    End Class
End Namespace

File: CS/PersonalOrganizer/ContactContextDataModel/DesignTime/DesignTimeContactContextUnitOfWork.cs
using System;
using System.Linq;
using System.Data;
using System.Data.Entity;
using System.Linq.Expressions;
using System.Collections.Generic;
using PersonalOrganizer.Common.Utils;
using PersonalOrganizer.Common.DataModel;
using PersonalOrganizer.Common.DataModel.EntityFramework;
using PersonalOrganizer.Model;

namespace PersonalOrganizer.ContactContextDataModel {
    public class ContactContextDesignTimeUnitOfWork : DesignTimeUnitOfWork, IContactContextUnitOfWork {
        static DesignTimeContactRepository contactsRepository = new DesignTimeContactRepository();

        public ContactContextDesignTimeUnitOfWork() {
        }

        bool IContactContextUnitOfWork.HasChanges() {
            return false;
        }
        IContactRepository IContactContextUnitOfWork.Contacts {
            get { return contactsRepository; }
        }
        public void Detach(object entity) {
        }
    }
}
